### DISCUSSION TITLE
Submit your Practicum exercise!

### DISCUSSION THREAD

**Discussion by onlyphantom** [onlyphantom/llm-python#19] [Practicum: Generative AI]
Create a copy of [the Generative AI Workshop template](https://colab.research.google.com/drive/1PjCGECqB37_TDD3kLHYl3IljlWxPNoXe?usp=sharing) (also in this [GitHub repo](https://github.com/onlyphantom/llm-python/blob/main/workshop/Generative_AI_Template_01.ipynb)).

You need to complete either Exercise 1 or Exercise 2 successfully. You can do both, but only the one with the best score is taken. Run all code and show all output in your Colab notebook before submission.

For Exercise 1:
* Implement the features based on the tips given to you (1 point)
* There should be no errors in running the functions (1 point)
* All 3 queries must have a satisfactory and correct answer (3 points)

For Exercise 2: 
* All 5 queries (`query_1`, ... `query_5`) must have a satisfactory and correct answer (5 points)

Scoring 4 points out of a possible 5 will warrant a certificate issued by Sectors. If you attempt both Exercises, only the one with the highest score is taken. 

You can discuss, and ask for help [in Discord](https://discord.com/invite/TAnZMmNS4X), but please do not share your answers or plagiarize. You are only cheating yourself off an education.

## Submission Format
To submit your completed exercise, open a new Discussion under 'Practicum: Generative AI', and paste a link to the completed work. This can be a .ipynb or a Google Colab notebook. The output **of each cell** must be shown, whether they are intermediary steps or not. Make sure to 'Run All' from top to bottom before saving your work.

Specify your name and first 4 letters of your email at the top of your notebook for me to identify you. 
Example: `Samuel Chan | samu` (if my email is indeed samuel@sectors.app)

> 🔴 IMPORTANT: DO NOT SHARE YOUR SECRETS. USE `from google.colab import userdata` to read your environment variables. Under no circumstance should you secret keys be ever visible. An inclusion of your key in the notebook will automatically disqualify your work, and for your own benefit, **I will promptly remove your thread here on GitHub so that your keys are not stolen**.

**Comment by dkarjadi** [comment]
Daniel Karjadi | daniel.karjadi@gmail.com
[Copy_of_Generative_AI_Template.zip](https://github.com/user-attachments/files/16453622/Copy_of_Generative_AI_Template.zip)

**Reply by onlyphantom** [reply]
Passed!

**Comment by JCaesarLG** [comment]
Abraham Juan | juancaesar204@gmail.com
Link to Github : https://github.com/JCaesarLG/GenAI_Assignment.git

**Reply by onlyphantom** [reply]
Not a bad attempt! But I suspect your AI will fail when the user query doesnt explicitly contain the dates ("last week", "yesterday", "first day of the month" etc). The original test is meant to help you build more robust AI that does well on those cases! 

Would you be keen on improving it? Your Gen AI model needs to be able to pass tests like these when date isnt explicitly provided! :)

```
queries = [
    "What are the top 5 companies by transaction volume on the first of this month?",
    "What are the most traded stock yesterday?",
    "What are the top 7 most traded stocks between 6th June to 10th June this year?"
]
```

**Reply by JCaesarLG** [reply]
Done, already updated the code to make the llm better understand the context of time. Sorry it took so long. I need to browse around to get a better understanding.

**Reply by onlyphantom** [reply]
passed!

**Comment by riki-profile** [comment]
Riki | riki
https://colab.research.google.com/drive/1wYP4t4r_BF9fwgVeMqJ0dnZjHDdx1orr?usp=sharing

**Reply by onlyphantom** [reply]
Passed!

**Comment by lahiardhan** [comment]
Muhammad Lahia Ardhan | lahi
https://colab.research.google.com/drive/1T-SbyqOI-nf04HYHqVVnjpYAvTOsAYHL?usp=sharing

**Reply by onlyphantom** [reply]
Passed!

**Comment by yggdrazil338** [comment]
Andry | andr
https://colab.research.google.com/drive/1Mb-tiodJusxnB90N8WyXRL_E150MS8LG#scrollTo=j9SUZx-2Zhi-

**Reply by onlyphantom** [reply]
Notebook is not public! Can you change the access to public for grading? :)

**Reply by yggdrazil338** [reply]
Apology. I have updated the access to public

**Reply by onlyphantom** [reply]
passed!

**Comment by alisulthan** [comment]
Ali Sulthan Abrar | alis
[Generative AI Sector_Ali Sulthan Abrar_Exersice.ipynb](https://colab.research.google.com/drive/10UIIupWDLH958XssgIuC3NDgzmuij-xT?usp=sharing)

**Reply by onlyphantom** [reply]
Passed!

**Comment by zcemklu** [comment]
Karen | kare | [GColab](https://colab.research.google.com/drive/1slzjUnJWgafzA4Gm_DKOfxmVqv-vECgP?usp=sharing)

**Reply by onlyphantom** [reply]
Well done! Passed!

**Comment by rianyan9723** [comment]
Rian Yan | rianyan9723@gmail.com 
Link: https://colab.research.google.com/drive/1-Omqv6vzoWwK1tfm3G9qu8mJYJrtf6ti?usp=sharing
